src/main/java/org/ow2/proactive/procci/service/occi/MixinService.java

```diff
@@ -186,10 +165,52 @@ public void addMixin(Mixin mixin) throws ClientException {
         }
     }
 
+    /**
+     * Remove a mixin from the database and update the references
+     * @param mixinTitle the title of the mixin to remove
+     */
+    public void removeMixin(String mixinTitle) {
+        Mixin mixinToRemove = getMixinByTitle(mixinTitle);
+        List<Entity> entities = mixinToRemove.getEntities();
+
+        cloudAutomationVariablesClient.delete(mixinTitle);
+
+        entities.forEach(entity -> {
+            getMixinNamesFromEntity(entity.getId()).remove(mixinTitle);
+        });
+
+        entities.forEach(entity -> {
+            Set<String> entityMixins = getMixinNamesFromEntity(entity.getId());
+            entityMixins.remove(mixinTitle);
+            cloudAutomationVariablesClient.update(entity.getId(), mapObject(entityMixins));
+        });
+
+    }
+
+    /**
+     *  Give a the name of each mixin
+     *
+     * @return a set containing the name of each mixin
+     */
     public Set<String> getProviderMixinsName() {
         return providerMixin.keySet();
     }
 
+    /**
+     * Give the list of the mixin name of an entity
+     *
+     * @param entityId is an occi entity
+     * @return the list of the mixin related to the entity instance
+     * @throws CloudAutomationServerException if there is an error in the cloud automation service response
+     */
+    Set<String> getMixinNamesFromEntity(String entityId) {
```

Comment:
This method already existed and was set as a package protection for test purpose.